### PR TITLE
Format commands help

### PR DESCRIPTION
**Pull request by theimowski**
This is supposed to auto generate command line help usage for each command, as well as auto generate the command line invocation in markdown.

It's still WIP, as some more commands are left to migrate to new approach

Feedback welcome

### CODE REVIEW

File: src/Paket/Commands.fs [hunk 1]
```diff
@@ -38,13 +38,23 @@ type AddArgs =
...
     interface IArgParserTemplate with
-        member __.Usage = ""
+        member this.Usage = 
+            match this with
+            | Nuget(_) -> "Nuget package id."
+            | Version(_) -> "Allows to specify version of the package."
+            | Project(_) -> "Allows to add the package to a single project only."
+            | Force -> "Forces the download and reinstallation of all packages."
+            | Interactive -> "Asks the user for every project if he or she wants to add the package to the projects's paket.references file."
+            | Hard -> "Replaces package references within project files even if they are not yet adhering to to Paket's conventions (and hence considered manually managed)."
```
Comment (mexx) [inline src/Paket/Commands.fs]: Please fix typo: `adhering to the Paket's`

File: src/Paket/Commands.fs [hunk 2]
```diff
@@ -98,14 +125,37 @@ type RemoveArgs =
...
 with 
     interface IArgParserTemplate with
-        member __.Usage = ""
+        member this.Usage =
+            match this with
+            | Nuget(_) -> "Nuget package id."
+            | Project(_) -> "Allows to add the package to a single project only."
+            | Force -> "Forces the download and reinstallation of all packages."
+            | Interactive -> "sks the user for every project if he or she wants to remove the package from the projects's paket.references file. By default every installation of the package is removed."
+            | Hard -> "Replaces package references within project files even if they are not yet adhering to to Paket's conventions (and hence considered manually managed)."
```
Comment (mexx) [inline src/Paket/Commands.fs]: Same typo here.

**Comment by mexx** [comment]
Good job @theimowski

**Comment by forki** [comment]
the red appyveyor test should be gone after rebase.

File: Paket.sln [hunk 3]
```diff
@@ -34,7 +34,6 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "content", "content", "{8E6D
 		docs\content\index.md = docs\content\index.md
 		docs\content\lock-file.md = docs\content\lock-file.md
 		docs\content\nuget-dependencies.md = docs\content\nuget-dependencies.md
-		docs\content\paket-config-file.md = docs\content\paket-config-file.md
```
Comment (forki) [inline Paket.sln]: is this gone?

Comment (theimowski) [inline Paket.sln]: yeah, I treat config as a command so it's now generated from the HelpTexts class.
Eventually we'll be able to fallback to separate md file for each command when this PR is merged

Comment (forki) [inline Paket.sln]: cool. just tell me when it's ready to merge - really cool stuff

Comment (theimowski) [inline Paket.sln]: thanks, I think it's ready now - I took it even a step further.

File: src/Paket/Commands.fs [hunk 4]
```diff
@@ -130,7 +198,12 @@ type PackArgs =
     | [<CustomCommandLine("releaseNotes")>] ReleaseNotes of string
 with
     interface IArgParserTemplate with
-        member __.Usage = ""
+        member this.Usage =
```
Comment (theimowski) [inline src/Paket/Commands.fs]: @mavnn could you please review the messages for this new command?

File: src/Paket/Commands.fs [hunk 5]
```diff
@@ -138,4 +211,77 @@ type PushArgs =
     | [<CustomCommandLine("apikey")>] ApiKey of string
 with
     interface IArgParserTemplate with
-        member __.Usage = ""
\ No newline at end of file
+        member this.Usage =
```
Comment (theimowski) [inline src/Paket/Commands.fs]: @mavnn - and the same here

File: src/Paket/Commands.fs (hunk 4 above)
Comment (mavnn) [inline src/Paket/Commands.fs]: Looks good

File: src/Paket/Commands.fs (hunk 5 above)
Comment (mavnn) [inline src/Paket/Commands.fs]: Looks good here too

**Comment by forki** [comment]
tumblr_inline_njhdq1yxtk1raprkq

**Comment by forki** [comment]
Just tested it. Looks great. Thanks so much.

**Comment by theimowski** [comment]
Thanks good to hear that

**Pull request merged**